### ISSUE TITLE
pcre64.dll missing on Windows (breaking the native feature)

### ISSUE THREAD

**Issue by eboth** [tridactyl/tridactyl#4538]
At some point (not sure if a Firefox or Tridactyl update had to do with it) the native messenger stopped working. I reinstalled it successfully, but it seems like whenever Tridactyl tries to use it, the only result is the following error message shown in the browser console:
```
stderr output from native app tridactyl: could not load: pcre64.dll
```
This happens e.g. when running `:source`, `:native` or `:! echo` – meaning for instance that I cannot load tridactylrc, which is a pain.

Is the native install supposed to install pcre64.dll? The file currently doesn't exist anywhere on my system, and the only other reference I've found to it was in the context of the Nim language.

OS: Windows 7 64 bit
Tridactyl version: 1.23.0pre6526-c663b8e2
Firefox version: 108.0.2 (64-Bit)
PATH variable (from outside Tridactyl as `:!` doesn't work in Tridactyl because of the problem described):
C:\Program Files\ConEmu\ConEmu\Scripts;C:\Program Files\ConEmu;C:\Program Files\ConEmu\ConEmu;C:\Program Files (x86)\Business Objects\Common\3.5\bin\NOTES\;C:\Program Files (x86)\Business Objects\Common\3.5\bin\NOTES\DATA\;C:\ProgramData\Oracle\Java\javapath;C:\Windows;C:\Windows\system32;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\Wbem;C:\Program Files (x86)\Windows Resource Kits\Tools;C:\Program Files\Mercurial;C:\Program Files\Microsoft SQL Server\110\Tools\Binn;C:\Program Files\Windows Imaging;c:\Program Files (x86)\Android\android-sdk\platform-tools;C:\php54;C:\ProgramData\ComposerSetup\bin;C:\Program Files (x86)\Skype\Phone;C:\Program Files\nodejs;C:\Program Files\ThinkPad\Bluetooth Software;C:\Program Files\ThinkPad\Bluetooth Software\syswow64;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\x86;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\x64;C:\ProgramData\chocolatey\bin;C:\Windows\System32\WindowsPowerShell\v1.0;systemroot%\System32\WindowsPow;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\nodejs\;C:\Program Files\dotnet\;C:\Python\python3\Scripts\;C:\Python\python3\;C:\Python\python3_32\;c:\python\Python2\;c:\mingw\bin;c:\php54;C:\Program Files (x86)\CVSNT;c:\Tools\Argyll\Argyll_V1.6.3\bin;c:\pear;c:\Program Files\WinRAR;C:\Ruby\bin;c:\Program Files (x86)\Safari\Apple Application Support\;c:\Program Files\Vim\vim\GVimExt64\;c:\Program Files\Vim\vim\

**Comment by bovine3dom** [comment]
I don't think you should need that dll separately, Nim should be packing everything into the one executable. I'd like to hear if more Windows users are having the same problem.

In the meantime you could use the old python native messenger which you can install with this script here: https://github.com/tridactyl/tridactyl/blob/master/native/win_install.ps1

**Comment by Briaoeuidhtns** [comment]
I'm having the same problem on windows 10

**Comment by bovine3dom** [comment]
https://github.com/nim-lang/Nim/issues/5541#issuecomment-286682924 suggests that it could be caused by 32 bit versions of `msvcrt.dll` on your path.

Could you try finding the 64-bit version of `msvcrt.dll` (maybe in C:\Windows ?) and putting it in the same folder as the executable which should be `%USERPROFILE%\.tridactyl\`?

**Comment by ELLIOTTCABLE** [comment]
I did a little digging on this, but ran out of time. One solution may be just to dump pcre entirely?

https://github.com/yglukhov/nimpy/issues/169#issuecomment-757142993

**Comment by bovine3dom** [comment]
Yeah we probably can, we're only using it here https://github.com/tridactyl/native_messenger/blob/f08e1286a54ecd2fad39f834b54f585843b3e51a/src/native_main.nim#L118

https://github.com/nitely/nim-regex isn't quite a drop-in replacement, we'd need to spend five minutes working out the difference between its `findAll` and the `findBounds` function we are using currently https://nim-lang.org/docs/re.html#findBounds%2Ccstring%2CRegex%2Cint%2Cint
